### PR TITLE
[BugFix] Fix compatability issue when ipv6 addr is used

### PR DESCRIPTION
**Pull request by nshangyiming**
## What I'm doing:
IPv4 addr is prefered by default when both IPv4 and IPv6 is used to maintain compatability.

Fixes SR-26089

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java [hunk 1]
```diff
@@ -67,9 +70,25 @@ public static List<InetAddress> getHosts() {
             Enumeration<InetAddress> a = e.getInetAddresses();
             while (a.hasMoreElements()) {
                 InetAddress addr = a.nextElement();
+                // IPv6 address starting with fe80 (Link-local Address) is not supported for now.
+                if (addr instanceof Inet6Address &&
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java:74]: 169.254.0.0/16 is a link-local ipv4 address, do we need to filter it out together?

File: fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java [hunk 2]
```diff
@@ -67,9 +70,25 @@ public static List<InetAddress> getHosts() {
...
+                        addr.getHostAddress().toLowerCase(Locale.ROOT).startsWith("fe80")) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java:75]: InetAddress has an interface `isLinkLocalAddress()`, looks good for the check?

File: fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java (hunk 1 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java:74]: didn't see the need for now

File: fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java (hunk 2 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java:75]: good idea, will fix it

File: fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java (hunk 1 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java]: what's the necessity to treat them differently?

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java]: not sure about the reason, for now, i only found the issue with ipv6 link local addr, ipv4 is fine

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/common/util/NetUtils.java]: so think it this way, what's the downside if filtering out ipv4 link-local address as well? our application will be more robust if the user's env happens to have a link-local ipv4 address bind to its interface, right?

And from the code, the `instanceof` check is no longer needed.

**Review by kevincai — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**